server.go

```diff
@@ -912,9 +912,17 @@ func (s *Server) handleRawConn(lisAddr string, rawConn net.Conn) {
 		return
 	}
 	rawConn.SetDeadline(time.Now().Add(s.opts.connectionTimeout))
+	ctx := context.Background()
+	for _, sh := range s.opts.statsHandlers {
+		ctx = sh.TagConn(ctx, &stats.ConnTagInfo{
+			RemoteAddr: rawConn.RemoteAddr(),
+			LocalAddr:  rawConn.LocalAddr(),
+		})
+		sh.HandleConn(ctx, &stats.ConnBegin{})
```

Comment:
This is a change in behavior, but it's subtle.

If the handshaker replaces the `conn` with something connected to another remote/local address, then that's not reflected here.

More importantly it doesn't match what we do later for remote/local addr, which is use the server transports addresses.  We probably want to move this after `NewServerTransport` and use `st.LocalAddr()` and `st.RemoteAddr()` instead.

(Also this change means we call the stats handler for every connection even if it doesn't finish handshaking correctly, which is probably also not what we want.)

Reply:
Hmmmm ok I was wondering whether I should do this after handshaking, I noticed it changed to log unconditionally. I remember you mentioned either passing conn around or just using the wrapper of conn to get addr info, and we chose latter.

Reply:
https://github.com/grpc/grpc-go/blob/master/internal/transport/http2_client.go#L372 moving it after transport will slightly change behavior as well, as previously TagConn was still called if there's a connection error in say writing intial http/2 preface/settings frame. For me, this is dependent on semantically what does TagConn represent? Accepting a TCP connection? Getting an HTTP/2 connection to a point where it finishes the initial frames?

Reply:
As discussed offline, waiting for handshaking before calling `TagConn` seems fine.

Reply:
Did this after newHTTP2Transport in new PR (still comes before first stats handler call after TagConn, so as we discussed offline, this is fine).